File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/solana/policies/create_spl_token_limits_policy.py
# Usage: uv run python solana/policies/create_spl_token_limits_policy.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv
from cdp.policies.types import (
    CreatePolicyOptions,
    SendSolanaTransactionRule,
    SplValueCriterion,
    MintAddressCriterion,
)

load_dotenv()

async def main():
    async with CdpClient() as cdp:
        # Create a policy that allows sending up to 1 USDC on Solana devnet
        policy = CreatePolicyOptions(
            scope="account",
            description="SPL Token Limits Policy",
            rules=[
                SendSolanaTransactionRule(
                    action="accept",
                    criteria=[
                        SplValueCriterion(
                            splValue="1000000",
                            operator="<=",
                        ),
                        MintAddressCriterion(
                            addresses=["4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU"],
                            operator="in",
                        ),
                    ],
                ),
            ],
        )

        result = await cdp.policies.create_policy(policy=policy)

        print("Created spl token limits policy: ", result.id)


asyncio.run(main())

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/solana/policies/createSplTokenLimitsPolicy.ts
// Usage: pnpm tsx solana/policies/createSplTokenLimitsPolicy.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

// Create a policy that allows sending up to 1 USDC on Solana devnet
const policy = await cdp.policies.createPolicy({
  policy: {
    scope: "account",
    description: "SPL Token Limits Policy",
    rules: [
      {
        action: "accept",
        operation: "sendSolTransaction",
        criteria: [
          {
            type: "splValue",
            splValue: "1000000",
            operator: "<=",
          },
          {
            type: "mintAddress",
            addresses: ["4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU"],
            operator: "in",
          },
        ],
      },
    ],
  },
});
console.log("Created spl token limits policy: ", policy.id);
